src/cli/cluster/base_path_proxy.js

```diff
@@ -53,6 +53,28 @@ export default class BasePathProxy {
     server.route({
       method: '*',
       path: `${basePath}/{kbnPath*}`,
+      config: {
+        pre: [
+          (req, reply) => {
+            map(clusterManager.workers, (worker) => {
+              if (worker.type === 'server' && !worker.listening && !worker.crashed) {
+                return fromNode(cb => {
+                  const done = () => {
```

Comment:
Not really. We need to react to the first event, 'listening' or 'crashed', and unbind the other. If they were both 'once' listeners, and the 'crashed' event happened first, then the 'listening' handler would never be unbound because the worker will never trigger 'listening' once it has crashed.